### ISSUE TITLE
When Cronet is enabled, gRPC should use grpc_cronet_plugin_registry.cc

### ISSUE THREAD

**Issue by muxi** [grpc/grpc#14701]

**Comment by markdroth** [comment]
Instead of using a whole different plugin registry, I think we should just use a channel arg to de-select the filters we don't want for cronet.  The channel arg can be checked in the function passed to `grpc_channel_init_register_stage()` for adding the filter.

**Comment by muxi** [comment]
Switching between chttp2 and Cronet happens at build time; that makes me feel more appropriate to use a different plugin registry for Cronet. We can drop the binaries of unused plugins.

If we are going to do that at run time it probably makes more sense to have an extra enum in [`channel_stack_type`](https://github.com/grpc/grpc/blob/dc46c7f565196d6684d5cf6ee81e99b413ba5282/src/core/lib/surface/channel_stack_type.h) for Cronet rather than filtering channel arg in every filter.

**Comment by markdroth** [comment]
I don't think it makes sense to have a different channel stack type for each transport.

If the goal here is to avoid building plugins that we don't need for Cronet, then presumably this is not just a plugin registration issue, but actually a build issue.  How is that build integration done?  For example, isn't the client_channel code also unused with Cronet?  If so, how do we avoid building the client_channel code for Cronet?

**Comment by muxi** [comment]
To avoid building plugins that we don't need for Cronet is the reason that I would like to use a separate plugin registration. I am not going to address that in this issue though. This particular issue is only about using the alternative plugin_registry file for Cronet and unblocks #14615. Making an additional channel arg would unblock #14615 too but diverts from the path of follow-up build integration changes.

**Comment by markdroth** [comment]
I see.  So today, we are actually building things we don't need for cronet, and the goal here is to fix that?

How are we going to indicate which plugins should be built for cronet and which should not?

I think @nicolasnoble should weigh in on this proposal.

**Comment by muxi** [comment]
> So today, we are actually building things we don't need for cronet

Yes

> and the goal here is to fix that

I think the goal here is to resolve the issue in #14615 in a way that would not be reverted when we want to improve the build integration.

> How are we going to indicate which plugins should be built for cronet and which should not?

Plugins built for cronet are specified by filegroups in `build.yaml`. Some reorganizations there are required (e.g. `client_channel` is not needed for Cronet but `grpc_transport_chttp2_client_secure` transitively depend on it). I'll leave that when we work on build integration changes.
